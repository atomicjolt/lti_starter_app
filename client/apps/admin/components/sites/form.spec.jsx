import React from 'react';
import TestRenderer from 'react-test-renderer';
import Form from './form';

describe('site form', () => {
  let result;
  let instance;
  let props;
  let modalClosed;

  beforeEach(() => {
    modalClosed = false;
    props = {
      setupSite: () => { modalClosed = true; },
      closeModal: () => { modalClosed = true; },
      onChange: () => {},
      isUpdate: false,
    };
    result = TestRenderer.create(<Form {...props} />);
    instance = result.root;
  });

  it('renders the form', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the Domain onclick event', () => {
    expect(modalClosed).toBeFalsy();
    const buttons = instance.findAllByType('button');
    const button = buttons.find(b => b.children[0] !== 'Cancel');
    button.props.onClick();
    expect(modalClosed).toBeTruthy();
  });

  describe('close modal', () => {
    it('closes', () => {
      expect(modalClosed).toBeFalsy();
      const buttons = instance.findAllByType('button');
      const button = buttons.find(b => b.children[0] === 'Cancel');
      button.props.onClick();
      expect(modalClosed).toBeTruthy();
    });
  });
});
