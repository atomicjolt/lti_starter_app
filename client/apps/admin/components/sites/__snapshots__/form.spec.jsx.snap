// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site form matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SiteForm
    closeModal={[Function]}
    isUpdate={false}
    onChange={[Function]}
    setupSite={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="o-grid o-grid__modal-top"
        >
          <div
            className="o-grid__item u-full"
          >
            <Input
              className="c-input"
              inputProps={
                Object {
                  "id": "site_url",
                  "name": "url",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                }
              }
              labelText="Canvas Domain"
            />
          </div>
          <div
            className="o-grid__item u-full"
          >
            <Input
              className="c-input"
              inputProps={
                Object {
                  "id": "site_oauth_key",
                  "name": "oauth_key",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                }
              }
              labelText="Canvas Developer ID"
            />
          </div>
          <div
            className="o-grid__item u-full"
          >
            <Input
              className="c-input"
              inputProps={
                Object {
                  "id": "site_oauth_secret",
                  "name": "oauth_secret",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                }
              }
              labelText="Canvas Developer Key"
            />
          </div>
        </div>,
        <button
          className="c-btn c-btn--yellow"
          onClick={[Function]}
          type="button"
        >
          Create
           Domain
        </button>,
        <button
          className="c-btn c-btn--gray--large u-m-right"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="o-grid__item u-full"
            >
              <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_url",
                    "name": "url",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Domain"
              />
            </div>,
            <div
              className="o-grid__item u-full"
            >
              <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_oauth_key",
                    "name": "oauth_key",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Developer ID"
              />
            </div>,
            <div
              className="o-grid__item u-full"
            >
              <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_oauth_secret",
                    "name": "oauth_secret",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Developer Key"
              />
            </div>,
          ],
          "className": "o-grid o-grid__modal-top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "url",
            "nodeType": "host",
            "props": Object {
              "children": <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_url",
                    "name": "url",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Domain"
              />,
              "className": "o-grid__item u-full",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "c-input",
                "inputProps": Object {
                  "id": "site_url",
                  "name": "url",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "labelText": "Canvas Domain",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "oauth_key",
            "nodeType": "host",
            "props": Object {
              "children": <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_oauth_key",
                    "name": "oauth_key",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Developer ID"
              />,
              "className": "o-grid__item u-full",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "c-input",
                "inputProps": Object {
                  "id": "site_oauth_key",
                  "name": "oauth_key",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "labelText": "Canvas Developer ID",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "oauth_secret",
            "nodeType": "host",
            "props": Object {
              "children": <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_oauth_secret",
                    "name": "oauth_secret",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Developer Key"
              />,
              "className": "o-grid__item u-full",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "c-input",
                "inputProps": Object {
                  "id": "site_oauth_secret",
                  "name": "oauth_secret",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "labelText": "Canvas Developer Key",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Create",
            " Domain",
          ],
          "className": "c-btn c-btn--yellow",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "Create",
          " Domain",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cancel",
          "className": "c-btn c-btn--gray--large u-m-right",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Cancel",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="o-grid o-grid__modal-top"
          >
            <div
              className="o-grid__item u-full"
            >
              <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_url",
                    "name": "url",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Domain"
              />
            </div>
            <div
              className="o-grid__item u-full"
            >
              <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_oauth_key",
                    "name": "oauth_key",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Developer ID"
              />
            </div>
            <div
              className="o-grid__item u-full"
            >
              <Input
                className="c-input"
                inputProps={
                  Object {
                    "id": "site_oauth_secret",
                    "name": "oauth_secret",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  }
                }
                labelText="Canvas Developer Key"
              />
            </div>
          </div>,
          <button
            className="c-btn c-btn--yellow"
            onClick={[Function]}
            type="button"
          >
            Create
             Domain
          </button>,
          <button
            className="c-btn c-btn--gray--large u-m-right"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="o-grid__item u-full"
              >
                <Input
                  className="c-input"
                  inputProps={
                    Object {
                      "id": "site_url",
                      "name": "url",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    }
                  }
                  labelText="Canvas Domain"
                />
              </div>,
              <div
                className="o-grid__item u-full"
              >
                <Input
                  className="c-input"
                  inputProps={
                    Object {
                      "id": "site_oauth_key",
                      "name": "oauth_key",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    }
                  }
                  labelText="Canvas Developer ID"
                />
              </div>,
              <div
                className="o-grid__item u-full"
              >
                <Input
                  className="c-input"
                  inputProps={
                    Object {
                      "id": "site_oauth_secret",
                      "name": "oauth_secret",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    }
                  }
                  labelText="Canvas Developer Key"
                />
              </div>,
            ],
            "className": "o-grid o-grid__modal-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "url",
              "nodeType": "host",
              "props": Object {
                "children": <Input
                  className="c-input"
                  inputProps={
                    Object {
                      "id": "site_url",
                      "name": "url",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    }
                  }
                  labelText="Canvas Domain"
                />,
                "className": "o-grid__item u-full",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "c-input",
                  "inputProps": Object {
                    "id": "site_url",
                    "name": "url",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "labelText": "Canvas Domain",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "oauth_key",
              "nodeType": "host",
              "props": Object {
                "children": <Input
                  className="c-input"
                  inputProps={
                    Object {
                      "id": "site_oauth_key",
                      "name": "oauth_key",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    }
                  }
                  labelText="Canvas Developer ID"
                />,
                "className": "o-grid__item u-full",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "c-input",
                  "inputProps": Object {
                    "id": "site_oauth_key",
                    "name": "oauth_key",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "labelText": "Canvas Developer ID",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "oauth_secret",
              "nodeType": "host",
              "props": Object {
                "children": <Input
                  className="c-input"
                  inputProps={
                    Object {
                      "id": "site_oauth_secret",
                      "name": "oauth_secret",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    }
                  }
                  labelText="Canvas Developer Key"
                />,
                "className": "o-grid__item u-full",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "c-input",
                  "inputProps": Object {
                    "id": "site_oauth_secret",
                    "name": "oauth_secret",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "labelText": "Canvas Developer Key",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Create",
              " Domain",
            ],
            "className": "c-btn c-btn--yellow",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            "Create",
            " Domain",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "className": "c-btn c-btn--gray--large u-m-right",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
